### PR TITLE
Add metadata plugin; update doc review process

### PR DESCRIPTION
**Pull request by brianhlin**
CC: @edquist @timtheisen @efajardo @ddavila0 @djw8605

### CODE REVIEW

File: docs/documentation/reviewing-documentation.md [hunk 1]
```diff
@@ -18,7 +19,18 @@ Content reviews are for validating the correctness of technical steps and detail
...
+[layout](/documentation/writing-documentation#document-layout) guidelines, as well as correct spelling and grammar.
 
-Editorial reviews are for ensuring docs meet our [style](/documentation/style-guide) and [layout](/documentation/writing-documentation#document-layout) guidelines; improving readability; and proofing spelling, grammar, and punctuation.
+Completing a Review
+-------------------
+
+Upon completion of the review:
+
+1.  Update (or add) the review date in the
+    [metadata](https://squidfunk.github.io/mkdocs-material/extensions/metadata/#redirecting-to-another-page) at the top
```
Comment (matyasselmeci) [inline docs/documentation/reviewing-documentation.md]: I didn't find this document to be helpful so IMO just skip the link.

What does the finished product look like?

**Comment by djw8605** [comment]
Is the `DateReviewed` displayed somewhere on the document?

**Comment by brianhlin** [comment]
With the metadata plugin, `KEY: VALUE` pairs above the h1 are not displayed in the resultant document.

**Comment by matyasselmeci** [comment]
So basically it's invisible?

**Comment by brianhlin** [comment]
Yeah, I just checked and it's not even hidden away in the generated HTML.

> That wasn't what I had in mind when I made the 'last reviewed' suggestion. I was thinking of something user-visible, like the way a wiki displays the last modified date, but with more meaning. This could give the admin some reassurance that the specific doc they're looking at is up-to-date... or not.

The plugin allows us to pass in metadata to the theme (that's why the [material theme documentation](https://squidfunk.github.io/mkdocs-material/extensions/metadata/) mentions specific keys), so we could add this feature in later. But I'm not interested in displaying that data, only using it internally to keep track of what docs are in need of review.

**Comment by matyasselmeci** [comment]
So will we have an automated notification for pages that get too old? Because it's not going to be useful if we have to go digging to find out -- any of _us_ can look at a page for 15 seconds and go 'oh my grid this is so wrong', it's the users that can't.

**Comment by brianhlin** [comment]
I'm sure we could develop a one-liner to find pages that haven't been reviewed in N months so I don't think this is something urgent enough to have regular notifications. And date since last review doesn't necessarily correlate with a page's correctness so I don't want to give folks the impression that a page is wrong just because it hasn't been updated recently. Since 1. this update includes some policy change about reviews and 2. doesn't actually change doc content, I'm going to merge it.

**Pull request merged**